### PR TITLE
Add support for defining collision margin data in SRDF

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR adds the ability to define collision margin data in the SRDF as shown below and gets loaded into the environment.

``` xml
<collision_margins default_margin="0.025">
  <pair_margin link1="link_6" link2="link_5" margin="0.01"/>
  <pair_margin link1="link_5" margin="0.015"/>
</collision_margins>
```

Additional change to accommodate this change:

- Move CollisionMarginData to tesseract_common
- Simplified SRDF removing setters and getters and exposed variables directly
- Add new environment command ChangeCollisionMarginsCommand and deprecated the other two command now that this one command encompasses both. They will be removed in about a week to give time for project to update.

### CODE REVIEW

File: tesseract_scene_graph/test/tesseract_scene_graph_srdf_unit.cpp [hunk 1]
```diff
@@ -1974,6 +2000,125 @@ TEST(TesseractSceneGraphSRDFUnit, SRDFGroupTCPsUnit)  // NOLINT
...
+  {  // missing pair margin
+    std::string str = R"(<robot name="abb_irb2400">
+                           <collision_margins default_margin="0.025">
+                             <pair_margin link1="link_6" link2="link_5" margin="0.01"/>
+                             <pair_margin link1="link_5" link2="link_4"/>
+                           </collision_margins>
+                         </robot>)";
+    EXPECT_ANY_THROW(is_failure(str));
+  }
+}
```
Comment (mpowelson) [inline tesseract_scene_graph/test/tesseract_scene_graph_srdf_unit.cpp:2120]: Possible additional tests

- Negative margin
- Fields exist but are empty

**Review by mpowelson — APPROVED** [review]
Looks reasonable to me. Do you want to add this to one of the SRDFs in tesseract_support just as an example/test?

File: tesseract_scene_graph/test/tesseract_scene_graph_srdf_unit.cpp (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_scene_graph/test/tesseract_scene_graph_srdf_unit.cpp:2120]: These are good additional tests. I will work on adding these.

**Comment by Levi-Armstrong** [comment]
> Looks reasonable to me. Do you want to add this to one of the SRDFs in tesseract_support just as an example/test?

Sure, I will add it the sphere test with a larger margin between the wrist joints and the sphere.

**Pull request merged**